libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -358,27 +351,23 @@ void ArrayUtils::copyArrayToMemory(ArrayType const& _sourceType, bool _padToWord
 			m_context << Instruction::SWAP1 << u256(32) << Instruction::ADD;
 			m_context << Instruction::SWAP1;
 		}
-		// convert length to size
-		if (baseSize > 1)
-			m_context << u256(baseSize) << Instruction::MUL;
+		if (!_sourceType.isByteArray())
+			convertLengthToSize(_sourceType);
 		// stack: <target> <source> <size>
 		m_context << Instruction::DUP1 << Instruction::DUP4 << Instruction::DUP4;
 		// We can resort to copying full 32 bytes only if
 		// - the length is known to be a multiple of 32 or
 		// - we will pad to full 32 bytes later anyway.
-		if (((baseSize % 32) == 0) || _padToWordBoundaries)
+		if (!_sourceType.isByteArray() || _padToWordBoundaries)
```

Comment:
is `!_sourceType.isByteArray()` implicitly checking for `the length is known to be a multiple of 32 or` as the comment still claims?

Reply:
I think the idea is to always pad to full 32 bytes during a copy to memory now, except for byte arrays, so the comment still applies and this explains the change below where you commented as well...